src/main/target/STM32F7X2/target.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdint.h>
+
+#include <platform.h>
+#include "drivers/io.h"
+#include "drivers/dma.h"
+
+#include "drivers/timer.h"
+#include "drivers/timer_def.h"
+
+const timerHardware_t timerHardware[USABLE_TIMER_CHANNEL_COUNT] = {
+
+    DEF_TIM(TIM12, CH2, PB15, TIM_USE_PWM | TIM_USE_PPM, 0, 0),
```

Comment:
Wouldn't it be better to not define any pins, so that a `diff` reflected the actual target config?

Reply:
This would become the entire list of available pins. Ultimately the TIM_USE would go... and it would simply be tag and index needed to find the timerHardware entry.

The timerhardware list will become the ultimate "lookup" of all possible combinations for timers.

Reply:
But `timerHardware` currently assigns one timer to each pin - this may or may not be the one that is used for a particular board, making a board's diff based config look inconsistent because it has to re-define timers for some pins, but not for others. Would it work to use placeholders to indicate 'undefined timer' in the unified targets?

Reply:
No this allows you to have multiple entries per pin in the array. I have added some for the STM32F7X2 target as an example.

Reply:
Ah I see. So the plan is to add all of the possible pin / timer combinations in here? This could probably be auto-generated as a one-off.

Reply:
It could be auto generated :) or people can add them as needed :)

Reply:
I'd prefer to have them auto-generated to start with. Adding them as needed whenever a new universal target is added or an existing one is converted will create a lot of churn, with the potential for bugs to be introduced and for the configuration for other targets to be invalidated, and the end result is that sooner or later all possible combinations will be in here anyway.

Reply:
I don't think there is an "auto generated" option. It's someone sitting down and going through the datasheet, and ensuring they are all covered.

Reply:
I thought they were all listed in `timer_def.h`, and could be extracted / converted from there?

But no matter how we do it, I think it should be done before target maintainers are starting to convert their targets.